### PR TITLE
Moved staff payment to review page, updated styles and have it captur…

### PR DESCRIPTION
**Pull request by jachurchill**
…ed by validation

*Issue #:* /bcgov/entity#6378

*Description of changes:* Moved staff payment to review page, updated styles and have it captured by validation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/common/StaffPayment.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <section id="staff-payment-section">
+  <section id="staff-payment">
```
Comment (severinbeauvais) [inline src/components/common/StaffPayment.vue:2]: The previous id followed our naming convention. Why did you change it?

Comment (jordiwes) [reply src/components/common/StaffPayment.vue:2]: This needs to match 
export enum ReviewSummaryFlags {
  'effective-date-time',
  'court-order',
  'certify-section',
  'staff-payment',
  'document-delivery-section'
}

is probably why :)

File: src/views/Alteration.vue [hunk 2]
```diff
@@ -73,6 +68,12 @@
         />
       </template>
 
+      <staff-payment
+        class="mt-10"
+        :validate="getAppValidate"
```
Comment (jordiwes) [inline src/views/Alteration.vue:73]: Doesn't look like the validate prop is used anywhere

File: src/store/mutations/mutations-model.ts [hunk 3]
```diff
@@ -207,6 +207,7 @@ export const mutateStaffPayment = (state: StateIF, staffPayment: StaffPaymentIF)
 
 export const mutateStaffPaymentValidity = (state: StateIF, validity: boolean) => {
   state.stateModel.staffPaymentStep.valid = validity
+  state.stateModel.newAlteration.validFlags.isValidStaffPayment = validity
```
Comment (severinbeauvais) [inline src/store/mutations/mutations-model.ts:210]: Don't change this right now... This comment is for all devs: this is a bit of a code smell...

I think we need to figure out our architecture for data that applies to different filings (like here - Corrections + Alterations). This is one approach but it results in duplicate code and won't scale well. I think a better approach would be to generalize data like this that is used in different filings. What do you guys think?

(I'm OK with this for now and we can deal with this if/when we add another filing to this project, but it would be nice to have an idea/plan what to do when we get there.)

File: src/components/common/StaffPayment.vue (hunk 1 above)
Comment (jachurchill) [reply src/components/common/StaffPayment.vue:2]: yea as @jordiwes said below this was to match with the ReviewSummaryFlags, would it be better for me to change the flag instead?

File: src/views/Alteration.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/views/Alteration.vue:73]: IMHO - leave it (less work). Future ticket (do you have the number?) can update it if needed.

File: src/components/common/StaffPayment.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/common/StaffPayment.vue:2]: Consistent naming is always good, but I see nothing consistent in the above list of flags!

If you have to edit this code for any other reason, please rename. Otherwise just leave it (less work).

**Comment by severinbeauvais** [comment]
Please update the app draft version number in package.json, then do `npm i`, then commit both package.json and the updated package-lock.json files.

File: src/views/Alteration.vue (hunk 2 above)
Comment (jachurchill) [reply src/views/Alteration.vue:73]: from the ticket 

> (to note - the Top Level red bar validation shown on comp 10a2 and auto scroll can be complete in the Top Level validation ticket #https://app.zenhub.com/workspaces/entity-5bf2f2164b5806bc2bf60531/issues/bcgov/entity/6546)
